### ISSUE TITLE
New Editor Modes - Pop Out Editor, bottom side editor, right side editor.

### ISSUE THREAD

**Issue by 5e-Cleric** [naturalcrit/homebrewery#2780]
### Your idea:

I think would be useful, and some users agree, to be able to decide where on your screen lays the editor , just like the console in most browsers behaves. 

This is not far away from having an independent editor that can pop out to a new window, so users can write in one screen, and see the rendered brew in another.

From the gitter chat:
> I did a Draft PR a while ago, for live updating to a local SharePage view of a brew
It's not a big stretch to see EditPage running the editor at 100% width

My guess for this feature would be having three working versions of the edit page, one with both brew and editor, one with only the brew, like a share page but with actual real time rendering, and a page with just the editor.

**Comment by G-Ambatte** [comment]
#2562 is the referenced PR for local live updates on SharePage.

I have previously thought about this as a useful feature for mobile users - with a button press, the user could dedicate their entire available screen space to the editor; press it again to toggle to full screen brew render. I imagine that this would be achieved by adding some state checks in SplitPane.

**Comment by dbolack-ab** [comment]
If we still consider this desirable, can we make this part of the Codemirror update?

**Comment by 5e-Cleric** [comment]
> If we still consider this desirable, can we make this part of the Codemirror update?

The behavior would be done completely outside of codemirror itself, but ideally in the splitpane or editPage component, so i guess it could be done, tagging @calculuschild to see how they like the idea of more steady work.

**Comment by calculuschild** [comment]
Yeah I don't think this would depend on codemirror update, but it is blocked by the edit page merge.
